File: Java/DesignPatterns/src/main/java/designPatterns/structural/facade/Facade.java
package designPatterns.structural.facade;

public class Facade {
  private SubClass.A subClassA;
  private SubClass.B subClassB;
  private SubClass.C subClassC;

  public Facade() {
    subClassA = new SubClass.A();
    subClassB = new SubClass.B();
    subClassC = new SubClass.C();
  }

  public String start() {
    return subClassA.doSomethingA() + ", " + subClassB.doSomethingB() + ", " + subClassC.doSomethingC();
  }
}

File: Kotlin/DesignPatterns/src/main/kotlin/structural/facade/Facade.kt
package structural.facade

class Facade {
  private var subClassA: SubClass.A? = null
  private var subClassB: SubClass.B? = null
  private var subClassC: SubClass.C? = null

  init {
    subClassA = SubClass.A()
    subClassB = SubClass.B()
    subClassC = SubClass.C()
  }

  fun start(): String {
    return "${subClassA?.doSomethingA()}, ${subClassB?.doSomethingB()}, ${subClassC?.doSomethingC()}"
  }
}